enterprise/views.py

```diff
@@ -192,6 +230,16 @@ def render_page_with_error_code_message(request, context_data, error_code, log_m
     )
 
 
+def should_upgrade_to_licensed_enrollment(consent_record, license_uuid):
+    """
+    In the event that a learner did enroll into audit from B2C and they consented
+    on the data sharing consent page, we want to upgrade their audit enrollment
+    into verified when they view the course in the learner portal and hit
+    the DSC GET endpoint again.
```

Comment:
Does this mean we end up with a side effecting GET? Same or similar to the one discussed here: https://edx-internal.slack.com/archives/CD3L7U2SZ/p1630002556001900?

Reply:
Yeah, it is.  This whole DSC flow is not great, particularly in how we hijacked it for licensed-enrollment.  It can be changed, but would require coordinated changes in frontend-app-learner-portal and license-manager.